### ISSUE TITLE
change the path 'https://huggingface.co/models' 

### ISSUE THREAD

**Issue by kaakang** [UTokyo-FieldPhenomics-Lab/DODA#7]
How do I change the path 'https://huggingface.co/models' to:' https://huggingface.co '？

**Comment by illrayy** [comment]
Thank you for your interest in our work.
Are you having problems downloading the model from huggingface? I can upload the required pre-trained model to the cloud storage

**Comment by kaakang** [comment]
Thanks for your reply, I have downloaded the model and solved the problem; In addition, can models trained in their own domain produce multi-class images like coco? (Forgive me if my question is silly, but I'm an expert at this.)

**Comment by illrayy** [comment]
That's a good question. We trained a model from scratch on the bdd dataset, the model can control the generated images with multiple classes like the coco dataset.
However, we found for multiple classes, additionally encoding the classes using the clip text encoding can achieve better control. To make the CLIP text encoder compatible with our domain encoder, it requires some modifications to `attention.py`.

**Comment by kaakang** [comment]
Thanks for your reply, so what can i do concretely to generate images with multiple of our domain?

**Comment by illrayy** [comment]
Well, it's a little bit complicated. Specifically, you first need to train ldm normally. After training, modify the cldm config file to define the CLIP text encoder as an additional conditional encoder. 
Then modify `ldm/models/diffusion/ddpm.py` accordingly to handle the additional input. Finally, add an attention layer after `self.attn2` of `CrossAttention` in `ldm/modules/attention.py` to integrate the class conditioninto the diffusion model through this attention layer.
In addition, you also need to refer to` utils/coco_dataset.py` to modify `utils/wheat_dataset.py` to read the classes correctly.

**Comment by kaakang** [comment]
Ok, I will try to do this as much as possible. In addition, why do I get an error after running train_wheate.py: "FileNotFoundError: [Errno 2] No such file or directory: 'models/control-wheat-ldm.ckpt'", the resume_path should not affect the running of the program.

Thanks again for taking the time to answer me!

**Comment by illrayy** [comment]
If you are training VAE or LDM, set `resume_path = None`. If you are training CLDM, `resume_path` is required.

**Comment by kaakang** [comment]
Thank you for your patience to explain, can it be multi-card training, if I can do what I should do.

**Comment by illrayy** [comment]
yes, you just need to modify `trainer = pl.Trainer()` in `train_wheat.py` 
For example, if you wanna use 4 gpus, you can set like this: 
`trainer = pl.Trainer(accelerator='ddp', gpus=4, precision=32, callbacks=[logger, checkpoint_callback], accumulate_grad_batches=accumulate_grad_batches, max_steps=max_steps)`

**Comment by kaakang** [comment]
> Well, it's a little bit complicated. Specifically, you first need to train ldm normally. After training, modify the cldm config file to define the CLIP text encoder as an additional conditional encoder. Then modify `ldm/models/diffusion/ddpm.py` accordingly to handle the additional input. Finally, add an attention layer after `self.attn2` of `CrossAttention` in `ldm/modules/attention.py` to integrate the class conditioninto the diffusion model through this attention layer. You also need to put the parameters of the new attention layer into the optimizer of `cldm/cldm.py` In addition, you also need to refer to` utils/coco_dataset.py` to modify `utils/wheat_dataset.py` to read the classes correctly.

ok. Thank you. In addition, can you tell me more details on this issue? I want to try to generate multi-category pictures in agriculture in the future.

**Comment by illrayy** [comment]
The content above describes the specific files and specific places that need modifications. It's difficult for me to further describe the specific codes using only words. After understanding the structure of the [ldm repository](https://github.com/CompVis/latent-diffusion), it should be relatively easy to make modifications based on the above content.
This repository is the simple implementation of our arxiv-version paper, which may change a lot after peer review. So currently I don't have plan to clean up and refine the code for public until the paper is finalised.

**Comment by kaakang** [comment]
Ok, thank you, your reply has been very helpful to me! Thanks again for your patience.
